src/string_pool.cr

```diff
@@ -70,26 +72,45 @@ class StringPool
   # pool.size # => 1
   # ```
   def get(str : UInt8*, len)
-    rehash if @size > 5 * @buckets.size
+    hash = hash(str, len)
+    get(hash, str, len)
+  end
 
-    index = bucket_index str, len
-    bucket = @buckets[index]
+  private def get(hash : UInt64, str : UInt8*, len)
+    rehash if @size >= @capacity / 4 * 3
 
-    if bucket
-      entry = find_entry_in_bucket(bucket, str, len)
-      if entry
-        return entry
+    mask = (@capacity - 1).to_u32
+    index, d = hash & mask, 1
```

Comment:
Could you split this into two assignments? Otherwise it looks a bit cryptic.

Also, could you explain here why `& mask` is needed? (I know, because later we see the comment about this, but maybe the algorithm can be explained somewhere?)

Reply:
Ok, will split.

Mask is just one of methods to get index to probe between 0 and capacity-1. It's fast and OK.